addons/sale/models/sale.py

```diff
@@ -187,7 +187,7 @@ def _search_invoice_ids(self, operator, value):
         required=True, readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]},
         domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]", tracking=1,
         help="If you change the pricelist, only newly added lines will be affected.")
-    currency_id = fields.Many2one("res.currency", related='pricelist_id.currency_id', string="Currency", readonly=True, required=True)
+    currency_id = fields.Many2one(related='pricelist_id.currency_id', store=True)
```

Comment:
shouldn't we set `depends=['pricelist_id']` to avoid a bad/lengthy recompute if someone changes the currency of the pricelist?

Reply:
otherwise it'll recompute currency on confirmed orders and lines... which will not go well

Reply:
also, readonly? because if someone adds this field in the view and changes it, they've just changed the pricelist's currency

Reply:
readonly=True by default on related fields ;)

Reply:
i forgot, in my head this was the case only for store=False but it seems you're right

Reply:
Ready for r+ then :pray: